btsensor/src/bthome/v1.rs

```diff
@@ -0,0 +1,657 @@
+    /// the actual value.
+    fn decimal_point(self) -> i32 {
+        match self {
+            Self::PacketId
+            | Self::Battery
+            | Self::HumidityShort
+            | Self::Count
+            | Self::Pm2_5
+            | Self::Pm10
+            | Self::Co2
+            | Self::Tvoc
+            | Self::MoistureShort
+            | Self::Timestamp
+            | Self::GenericBoolean
+            | Self::PowerOn
+            | Self::Open
+            | Self::BatteryLow
+            | Self::BatteryCharging
+            | Self::CarbonMonoxideDetected
+            | Self::Cold
+            | Self::Connected
+            | Self::DoorOpen
+            | Self::GarageDoorOpen
+            | Self::GasDetected
+            | Self::HeatAbnormal
+            | Self::LightDetected
+            | Self::Unlocked
+            | Self::Wet
+            | Self::MotionDetected
+            | Self::Moving
+            | Self::OccupancyDetected
+            | Self::PluggedIn
+            | Self::Home
+            | Self::Problem
+            | Self::Running
+            | Self::Safe
+            | Self::SmokeDetected
+            | Self::Sound
+            | Self::Tamper
+            | Self::VibrationDetected
+            | Self::WindowOpen
+            | Self::ButtonEvent
+            | Self::DimmerEvent => 0,
+            Self::Temperature
+            | Self::Humidity
+            | Self::Pressure
+            | Self::Illuminance
+            | Self::MassKg
+            | Self::MassLb
+            | Self::Dewpoint
+            | Self::Power
+            | Self::Moisture => 2,
+            Self::Energy | Self::Voltage | Self::Acceleration | Self::Gyroscope => 3,
+        }
+    }
+
+    fn is_boolean(self) -> bool {
+        matches!(
+            self,
+            Self::GenericBoolean
+                | Self::PowerOn
+                | Self::Open
+                | Self::BatteryLow
+                | Self::BatteryCharging
+                | Self::CarbonMonoxideDetected
+                | Self::Cold
+                | Self::Connected
+                | Self::DoorOpen
+                | Self::GarageDoorOpen
+                | Self::GasDetected
+                | Self::HeatAbnormal
+                | Self::LightDetected
+                | Self::Unlocked
+                | Self::Wet
+                | Self::MotionDetected
+                | Self::Moving
+                | Self::OccupancyDetected
+                | Self::PluggedIn
+                | Self::Home
+                | Self::Problem
+                | Self::Running
+                | Self::Safe
+                | Self::SmokeDetected
+                | Self::Sound
+                | Self::Tamper
+                | Self::VibrationDetected
+                | Self::WindowOpen
+        )
+    }
+}
+
+/// Attempts to decode the given service data as a BTHome v1 advertisement.
+pub fn decode(mut data: &[u8]) -> Result<Vec<Element>, DecodeError> {
```

Comment:
(optional suggestion) move this onto Element and call it decode_all()?

Reply:
Done.